roles/terraform/tasks/setup_vars.yml

```diff
@@ -40,38 +40,43 @@
 #     second the hypervisor (where KVM is installed)
 #####
 - name: local ansible, remote terraform, remote_2 kvm
   set_fact:
-    provider_uri: "qemu+ssh://{{ hostvars['terraform_node']['terraform_target_hypervisor_username'] | default(ssh_user,boolean=True) }}@{{ hostvars['terraform_node']['terraform_target_hypervisor'] }}/system"
-    ansible_jump_hosts: "{{ [ {'user':hostvars['terraform_node']['ansible_user'] | default(ssh_user,boolean=True), 'host':hostvars['terraform_node']['terraform_target_hypervisor'], 'port':hostvars['terraform_node']['terraform_target_hypervisor_port'] | default(ssh_port,boolean=True)  } ] + [ {'user':ansible_user, 'host':hostvars['terraform_node']['ansible_host'], 'port':hostvars['terraform_node']['ansible_port'] | default(22,boolean=True)  } ]  }}"
+    provider_uri: "qemu+ssh://{{ hostvars[hypervisor]['ansible_user'] | default(ssh_user,boolean=True) }}@{{ hostvars[hypervisor]['ansible_host'] }}/system"
+    ansible_jump_hosts: "{{ [ {'user':hostvars['terraform_node']['ansible_user'] | default(ssh_user,boolean=True), 'host':hostvars[hypervisor]['ansible_host'], 'port':hostvars[hypervisor]['ansible_port'] | default(ssh_port,boolean=True)  } ] + [ {'user':ansible_user, 'host':hostvars['terraform_node']['ansible_host'], 'port':hostvars['terraform_node']['ansible_port'] | default(22,boolean=True)  } ]  }}"
     terraform_bastion_enabled: True
-    terraform_bastion_host: "{{ hostvars['terraform_node']['terraform_target_hypervisor'] }}"
-    terraform_bastion_password: "{{ hostvars['terraform_node']['terraform_target_hypervisor_password'] | default(ssh_password,boolean=True) }}"
-    terraform_bastion_port: "{{ hostvars['terraform_node']['terraform_target_hypervisor_port'] | default(ssh_port,boolean=True) }}"
-    terraform_bastion_user: "{{ hostvars['terraform_node']['terraform_target_hypervisor_username'] | default(ssh_user,boolean=True) }}"
+    terraform_bastion_host: "{{ hostvars[hypervisor]['ansible_host'] }}"
+    terraform_bastion_password: "{{ hostvars[hypervisor]['ansible_ssh_pass'] | default(ssh_password,boolean=True) }}"
+    terraform_bastion_port: "{{ hostvars[hypervisor]['ansible_port'] | default(ssh_port,boolean=True) }}"
+    terraform_bastion_user: "{{ hostvars[hypervisor]['ansible_user'] | default(ssh_user,boolean=True) }}"
   when:
     - hostvars['terraform_node']['ansible_connection'] != 'local'
-    - hostvars['terraform_node']['terraform_target_hypervisor'] is defined and hostvars['terraform_node']['terraform_target_hypervisor'] != hostvars['terraform_node']['ansible_host']
+    - hostvars[hypervisor]['ansible_host'] is defined and hostvars[hypervisor]['ansible_host'] != hostvars['terraform_node']['ansible_host']
+  delegate_to: terraform_node
+
+- name: Set HCL Deploy path
+  set_fact:
+    hcl_deploy_path: "{{ lookup('env','HOME') }}/.terrible/{{ ansible_inventory_sources[0].split('/')[-1] }}/{{ provider_uri.split('/')[-2] }}"
```

Comment:
Is there another way to retrieve the variable, instead of use `{{ provider_uri.split('/')[-2] }}`?
Maybe using some Ansible filter, to improve the readability of the code.

Reply:
Will try to use jinja2 filters but array manipulation is mainly dealt with python usually

Reply:
Sadly there seems no better way to do this